src/utils/generator.js

```diff
@@ -1042,6 +1043,18 @@ export class Generator {
     this._entries.push(new TemporalBuildNodeEntry(entry));
   }
 
+  purgeEntriesWithGeneratorDepencies(): void {
```

Comment:
The kind of entries that get purged seem quite arbitrary. Can you add some comments that reflect your unhappiness with the situation? By itself, this function looks very authoritative.

Reply:
Added some scary looking comments.